pkg/resource/obcluster_task.go

```diff
@@ -141,11 +143,12 @@ func (m *OBClusterManager) ModifyOBZoneReplica() error {
 	return nil
 }
 
-func (m *OBClusterManager) DeleteOBZone() error {
+func (m *OBClusterManager) getZonesToDelete() ([]v1alpha1.OBZone, error) {
+	deletedZones := make([]v1alpha1.OBZone, 0)
 	obzoneList, err := m.listOBZones()
 	if err != nil {
 		m.Logger.Error(err, "List obzone failed")
-		return errors.Wrapf(err, "List obzone of obcluster %s failed", m.OBCluster.Name)
+		return deletedZones, errors.Wrapf(err, "List obzone of obcluster %s failed", m.OBCluster.Name)
```

Comment:
why not return (nil, err) here :)

Reply:
this function returns []v1alpha1.OBZone, err, not *[]v1alpha1.OBZone, err

Reply:
yes, slice type is actually a pointer type, it should be ok to return `nil`. The following test case passes.

```go
It("Empty Slice", Label("misc"), func() {
	var a []int
	Expect(len(a)).To(BeEquivalentTo(0))
	Expect(a).To(BeEmpty())
	Expect(a).To(BeNil())
})
```